invenio_communities/generators.py

```diff
@@ -158,6 +159,9 @@ def communities(self, identity):
 
     def needs(self, record=None, community_id=None, **kwargs):
         """Enabling Needs."""
+        if record is None:
```

Comment:
why was this needed?

Reply:
I removed it now, I placed it there to try and use this function also when creating a new community. We have this implementation elsewhere in the code, checking if the object being passed is empty, hence new.
Now it is not required, because I am using a different permission for creating a community and mofdifying one.